### PR TITLE
blog for file upload example for V5

### PR DESCRIPTION
**Pull request by caoxu2000**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by wswebcreation** [comment]
Hi @caoxu2000 

This might also help, I did an investigation for a customer of us and created [this](https://github.com/saucelabs-sample-test-frameworks/WebdriverIO-download-upload) project.

Currently I can't migrate it to V5 because there is a bug in the [@wdio/firefox-profile-service](https://github.com/webdriverio/webdriverio/issues/3297), but the rest wouldn't change.

File: website/blog/file-upload-example.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: File Upload Example in V5
+author: Xu Cao
+authorURL: https://twitter.com/xucao
+authorImageURL: https://pbs.twimg.com/profile_images/1085903692673384448/McZrlD3Q_400x400.jpg
+---
+
+We got so many questions in gitter.im regarding the file upload in V5. So this is my first attempt to
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion
We got so many questions in [our support channel on Gitter](https://gitter.im/webdriverio/webdriverio) regarding the file upload in V5. So this is my first attempt to 
```

File: website/blog/file-upload-example.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+The markup is this:
+
+```js
+// Two hidden input elements, 1 for single file upload 1 for multiple.
+
+<input class="upload-data-file-input hidden" type="file">
+<input class="upload-data-file-input-multiple hidden" type="file">
+
+```
+<br>
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion

```

File: website/blog/file-upload-example.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+Here is the code to 1. Make the hidden element visible first 2. Wait it displayed 3. setValue(filePath) on it.
+That is essentially submit the form with the file path on local or a url:
+
+```js
+// normal Nodejs stuff to create a variable pointing to the file
+
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion
```

File: website/blog/file-upload-example.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+const path = require('path');
+const filePath = path.join(__dirname, 'path/to/your/file');
+
+// The css class name "upload-data-file-input hidden" is just an example and you can replace with your app.
+
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion
```

File: website/blog/file-upload-example.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+browser.execute('document.getElementsByClassName("upload-data-file-input hidden")[0].style.display = "block"');
+$('.upload-data-file-input').waitForDisplayed();
+$('.upload-data-file-input').setValue(filePath);
+
+```
+<br>
+
+That was it!!! Happy Uploading 😉🙌🏻 ❤️<br>
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion
That was it!!! Happy Uploading 😉🙌🏻 ❤️
```

File: website/blog/file-upload-example.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
+title: File Upload Example in V5
+author: Xu Cao
+authorURL: https://twitter.com/xucao
+authorImageURL: https://pbs.twimg.com/profile_images/1085903692673384448/McZrlD3Q_400x400.jpg
+---
+
+We got so many questions in gitter.im regarding the file upload in V5. So this is my first attempt to 
+address that. Sorry in advance that this solution only tested in Chrome browser with selenium-standalone
+service.<br>
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion
service.
```

File: website/blog/file-upload-example.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+<br>
```
Comment (christian-bromann) [inline website/blog/file-upload-example.md]:
```suggestion

```

**Comment by wswebcreation** [comment]
@christian-bromann 

This link in my comment is already linking to the issue I created a few days ago, but here it is again https://github.com/webdriverio/webdriverio/issues/3297

**Comment by diachedelic** [comment]
Perhaps a more elegant solution than manually displaying a hidden `<input type="file">` is to use `elementSendKeys`?  I had success with this custom command:

```
browser.addCommand('chooseFile', function(filePath) {
  this.elementSendKeys(this.elementId, filePath.split(''))
}, true)
```

File: website/blog/file-upload-example.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+That is essentially submit the form with the file path on local or a url:
+
+```js
+// normal Nodejs stuff to create a variable pointing to the file
+
+const path = require('path');
+const filePath = path.join(__dirname, 'path/to/your/file');
+
+// The css class name "upload-data-file-input hidden" is just an example and you can replace with your app.
+browser.execute('document.getElementsByClassName("upload-data-file-input hidden")[0].style.display = "block"');
```
Comment (tnypxl) [inline website/blog/file-upload-example.md]: Is there a reason you aren't passing in the located web element as an argument?

**Comment by christian-bromann** [comment]
Closing in favor of #3632

**Pull request closed without merging**